envoy/tcp/upstream.h

```diff
@@ -78,10 +78,13 @@ class GenericConnectionPoolCallbacks {
    * Called to indicate a failure for GenericConnPool::newStream to establish a stream.
    *
    * @param reason supplies the failure reason.
+   * @param failure_reason failure reason string (Note: it is expected that the caller will provide
+   * matching `reason` and `failure_reason`).
    * @param host supplies the description of the host that caused the failure. This may be nullptr
    *             if no host was involved in the failure (for example overflow).
    */
   virtual void onGenericPoolFailure(ConnectionPool::PoolFailureReason reason,
+                                    absl::string_view failure_reason,
```

Comment:
if we're not adding details if we fail the CONNECT string I think we need to rename all the failure reason to transport_failure_reason

Reply:
Hi, is there a specific definition to the transport failure? IMO, all errors except the errors caused by normal requests/responses are transport error. Maybe I've been wrong all along.

Reply:
I thought this was a catch-all `failure reason`  in case of any failure in connection pool.  Since `StreamInfo::setUpstreamTransportFailure` is specific to a transport failure, I was not sure if we can introduce a new failure reason here and have it propagate as a `transport failure`.  I don't have a strong opinion on this :). Happy to make a change that would be most appropriate.